tensorflow/contrib/cmake/tf_tests.cmake

```diff
@@ -0,0 +1,384 @@
+enable_testing()
+
+#
+# get a temp path for test data
+#
+function(GetTestRunPath VAR_NAME OBJ_NAME)
+    if(WIN32) 
+      if(DEFINED ENV{TMP}) 
+        set(TMPDIR "$ENV{TMP}")
+      elseif(DEFINED ENV{TEMP})
+        set(TMPDIR "$ENV{TEMP}")
+      endif()
+      string(REPLACE "\\" "/" TMPDIR ${TMPDIR})
+    else() 
+      set(TMPDIR "$ENV{TMPDIR}") 
+    endif() 
+    if(NOT EXISTS "${TMPDIR}")
+       message(FATAL_ERROR "Unable to determine a path to the temporary directory")
+    endif()
+    set(${VAR_NAME} "${TMPDIR}/${OBJ_NAME}" PARENT_SCOPE)
+endfunction(GetTestRunPath)
+
+#
+# create test for each source
+#
+function(AddTests)
+  cmake_parse_arguments(_AT "" "" "SOURCES;OBJECTS;LIBS;DATA;DEPENDS" ${ARGN})
+  foreach(sourcefile ${_AT_SOURCES})
+    string(REPLACE "${tensorflow_source_dir}/" "" exename ${sourcefile})
+    string(REPLACE ".cc" "" exename ${exename})
+    string(REPLACE "/" "_" exename ${exename})
+    AddTest(
+      TARGET ${exename}
+      SOURCES ${sourcefile}
+      OBJECTS ${_AT_OBJECTS}
+      LIBS ${_AT_LIBS}
+      DATA ${_AT_DATA}
+      DEPENDS ${_AT_DEPENDS}
+    )
+  endforeach()
+endfunction(AddTests)
+
+#
+# create once test
+#
+function(AddTest)
+  cmake_parse_arguments(_AT "" "TARGET" "SOURCES;OBJECTS;LIBS;DATA;DEPENDS" ${ARGN})
+  
+  list(REMOVE_DUPLICATES _AT_SOURCES)
+  list(REMOVE_DUPLICATES _AT_OBJECTS)
+  list(REMOVE_DUPLICATES _AT_LIBS)
+  if (_AT_DATA)
+    list(REMOVE_DUPLICATES _AT_DATA)
+  endif(_AT_DATA)
+  if (_AT_DEPENDS)
+    list(REMOVE_DUPLICATES _AT_DEPENDS)
+  endif(_AT_DEPENDS)
+  
+  add_executable(${_AT_TARGET} ${_AT_SOURCES} ${_AT_OBJECTS})
+  target_link_libraries(${_AT_TARGET} ${_AT_LIBS})
+
+  GetTestRunPath(testdir ${_AT_TARGET})
+  set(tempdir "${testdir}/tmp")
+  file(REMOVE_RECURSE "${testdir}")
+  file(MAKE_DIRECTORY "${testdir}")
+  file(MAKE_DIRECTORY "${tempdir}")
+  add_test(NAME ${_AT_TARGET} COMMAND ${_AT_TARGET} WORKING_DIRECTORY "${testdir}")
+  set_tests_properties(${_AT_TARGET}
+    PROPERTIES ENVIRONMENT "TEST_TMPDIR=${tempdir};TEST_SRCDIR=${testdir}"
+  )
+
+  foreach(datafile ${_AT_DATA})
+    add_custom_command(
+      TARGET ${_AT_TARGET} POST_BUILD
+      COMMAND ${CMAKE_COMMAND} -E copy
+             "${CMAKE_CURRENT_SOURCE_DIR}/${datafile}"
+             "${testdir}/${datafile}"
+      DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/${datafile}"
+    )
+  endforeach()
+
+  if (_AT_DEPENDS)
+    add_dependencies(${_AT_TARGET} ${_AT_DEPENDS})
+  endif()
+endfunction(AddTest)
+
+#
+# create python test for each script
+#
+function(AddPythonTests)
+  cmake_parse_arguments(_AT "" "" "SOURCES;DATA;DEPENDS" ${ARGN})
+  list(REMOVE_DUPLICATES _AT_SOURCES)
+  if (_AT_DATA)
+    list(REMOVE_DUPLICATES _AT_DATA)
+  endif(_AT_DATA)
+  if (_AT_DEPENDS)
+    list(REMOVE_DUPLICATES _AT_DEPENDS)
+  endif(_AT_DEPENDS)
+  
+  foreach(sourcefile ${_AT_SOURCES})
+    add_test(NAME ${sourcefile} COMMAND python ${sourcefile})
```

Comment:
Instead of directly calling Python, can we use a variable to define which python binary to use?
Like here:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/cmake/tf_core_framework.cmake#L181

in Anaconda3 installations python binary location did not end up in PATH. I think this will be more convenient, as it is the same way we are doing for cmake itself.